### PR TITLE
Handle missing combination

### PR DESCRIPTION
**Pull request by swiftgist**
The ceph.conf will have rgw sections if
  any minion has the rgw role OR
  rgw_configurations is set and minions have those roles

Signed-off-by: Eric Jackson <ejackson@suse.com>

### CODE REVIEW

File: srv/salt/ceph/configuration/files/ceph.conf.rgw [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [ client.{{ client }} ]
 rgw frontends = "civetweb port=80"
-rgw dns name = {{ grains['fqdn'] }}
+rgw dns name = {{ fqdn }}
```
[the remark this answers is not in the document]
Comment (swiftgist) [reply srv/salt/ceph/configuration/files/ceph.conf.rgw]: the fqdn comes from the for loop in the ceph.conf.j2 which is returned from the runner and matches the grain for that minion of that role

[the remark this answers is not in the document]
Comment (smithfarm) [reply srv/salt/ceph/configuration/files/ceph.conf.rgw]: I've seen the python functions that allegedly return the short hostname return the FQDN, but I've never seen the fqdn function return the short hostname. (Just my limited experience.)

File: srv/modules/runners/select.py [hunk 2]
```diff
@@ -128,20 +128,28 @@ def from_(pillar, *args, **kwargs):
 
     local = salt.client.LocalClient()
     search = "I@roles:master"
-    result = local.cmd(search , 'pillar.get', [ pillar ], expr_form="compound")
+    try:
+        roles = local.cmd(search , 'pillar.get', [ pillar ], expr_form="compound").values()[0].keys()
+    except AttributeError as err:
```
[the remark this answers is not in the document]
Comment (swiftgist) [reply srv/modules/runners/select.py]: It's the only one I encountered currently.  If local.cmd fails for any reason, then something is wrong and I really don't want a stack trace in the middle of a ceph.conf.  How generic an error do you want to use?

**Comment by smithfarm** [comment]
Test running: http://149.202.174.158:8081/ubuntu-2017-09-01_14:38:15-deepsea:basic:rgw-ses5---basic-openstack/

File: srv/modules/runners/select.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (theanalyst) [reply srv/modules/runners/select.py]: well a general 
`except:
    roles=  []
` 
would catch everything and return an empty roles instead,

File: srv/salt/ceph/configuration/files/ceph.conf.rgw (hunk 1 above)
[the remark this answers is not in the document]
Comment (swiftgist) [reply srv/salt/ceph/configuration/files/ceph.conf.rgw]: It's not a function.... it's a variable.

File: srv/modules/runners/select.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (swiftgist) [reply srv/modules/runners/select.py]: changed.

**Comment by smithfarm** [comment]
**running** http://149.202.174.158:8081/ubuntu-2017-09-01_14:56:55-deepsea:basic-ses5---basic-openstack/

**Pull request merged**